### PR TITLE
PR for #20: Create framework to handle external inputs

### PR DESCRIPTION
**Pull request by lucamlouzada**
**Closes** #20.

**Changes and deliverables:**
In issue #20, I introduced a new approach for handling external file dependencies. It is quite straightforward: a file called [`make_externals.sh`](https://github.com/gentzkow/GentzkowLabTemplate/blob/c22d4446fd3549bdaef9c9dda09fc1a758e0c1b4/lib/shell/make_externals.sh) creates symlinks based on what is specified in `local_env.sh`. I have edited the [`local_env_template.sh`](https://github.com/gentzkow/GentzkowLabTemplate/blob/c22d4446fd3549bdaef9c9dda09fc1a758e0c1b4/lib/setup/local_env_template.sh) for this reason. Note that I have also deleted the `local_env.sh` from the template per https://github.com/gentzkow/GentzkowLabTemplate/issues/20#issuecomment-2384365857.

Note that in #26 we are currently discussing the creation of `setup.sh`. The idea is that this `make_externals.sh` will be called by `setup.sh`, or it can be run on its own. Details about these changes and the discussion about them can be found in the issue thread.

**Review**
Other than reviewing the changes and testing the overall logic of the new approach, it'd be great to test the robustness for errors in the spirit of #19. For example, testing what happens when invalid paths are listed, when there are no paths to add, and so on.

I am requesting a review from @ShiqiYang2022. Please let me know if you have any questions. Thanks!

### CODE REVIEW

File: lib/shell/make_externals.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+set -e
+
+MAKE_EXTERNALS_SCRIPT_DIR=$(dirname "$(realpath "${BASH_SOURCE[0]}")")
+REPO_ROOT=$(realpath "$MAKE_EXTERNALS_SCRIPT_DIR/../../")
+
+source "${REPO_ROOT}/local_env.sh"
+
+# Check if there are valid paths in local_env
+create_links=false
+for target_path in "${EXTERNAL_PATHS[@]}"; do
```
Comment (ShiqiYang2022) [inline lib/shell/make_externals.sh:11]: I think the error message "Target path $target_path does not exist for $name" will only populated up if there are both at least one valid path for external target and both at least one wrong path.

What if all `target_path`(s) in `local_env.sh` don't exist? I think then `create_links` will return `false`, but program will exit without any error message. I suggest also add error/warning message handling for this case.

**Review by ShiqiYang2022 — CHANGES_REQUESTED** [review]
Thanks @lucamlouzada for the great work! I have tested on my end and all looks good to me with only one exception which I left as a comment. After we resolve that issue I am then willing to sign off.

File: lib/shell/make_externals.sh (hunk 1 above)
Comment (lucamlouzada) [reply lib/shell/make_externals.sh:11]: Thanks @ShiqiYang2022 , that's a good point. I have pushed a change to the if-statement logic in 8913f7e which I think will solve this.

Comment (ShiqiYang2022) [reply lib/shell/make_externals.sh:11]: @lucamlouzada That looks perfect, thanks!

**Review by ShiqiYang2022 — APPROVED** [review]
All the opening issues from me are properly addressed so I am approving, thanks a lot for the work @lucamlouzada!

**Pull request merged**